torchtune/utils/memory.py

```diff
@@ -23,6 +23,40 @@
 ACWrapPolicyType: Type = Union[Set[Type], Callable[[nn.Module, bool, int], bool]]
 
 
+def get_ac_policy(model_type, modules_to_wrap: Set[Type]) -> ACWrapPolicyType:
+    """
+    Get the activation checkpointing policy for the model.
+
+    Args:
+        model_type (str): Model type.
+        modules_to_wrap (Set[Type]): Set of module types to wrap.
+
+    Returns:
+        ACWrapPolicyType: Policy to wrap module.
+    """
+    if str(model_type) == "LLAMA3":
+        # ensure torch.nn.Embedding activations are checkpointed.
+        modules_to_wrap.add(torch.nn.Embedding)
+        return partial(_llama3_ac_policy, modules_to_wrap=modules_to_wrap)
+    else:
+        return ModuleWrapPolicy(modules_to_wrap)
+
+def _llama3_ac_policy(module: nn.Module, recurse: bool, modules_to_wrap, **kwargs):
+    # Label that output_proj should be wrapped individually.
+    if isinstance(module, modules.TransformerDecoder):
+        targ = module.output.module
```

Comment:
this should probably use a helper function called `_get_fsdp_wrapped_module` or something that is aware of whether `module.output` is wrapped in FSDP or not and intelligently unwraps it.